lib/finch/pool/round_robin.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Finch.Pool.RoundRobin do
+  @moduledoc """
+  Cycle through the pools one by one, in a consistent order.
+  """
+  @behaviour Finch.Pool.Strategy
+
+  @impl true
+  def registry_value(%{count: count}) do
+    counter = :counters.new(1, [:atomics])
+    %{strategy: __MODULE__, counter: counter, count: count}
+  end
+
+  @impl true
+  def handle_enqueue(%{counter: counter}) do
+    :counters.add(counter, 1, 1)
+  end
+
+  @impl true
+  def choose_pool([{_, %{counter: counter, count: count}} | _] = pids) do
+    index = :counters.get(counter, 1)
```

Comment:
Unfortunately I think this does not really guarantee round robin, because you can have handle_enqueue being called multiple times before you get to choose the pool. Or you choose the pool multiple times before handle_enqueue is called and bumped. You think you would need to use ets:update_counter if you want an actual round robin.

Reply:
Ahh right of course :( Ok, I'll switch to the ets counter. Thanks for pointing that out!

Reply:
You can probably use the `atomics` module that has a `add_get` function

Reply:
That's right! I try that out